lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -428,21 +420,45 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
         }
     }
 
-    handleGeneralInfoPanelState() {
-        this.generalInfoPanelState = !this.generalInfoPanelState;
+    toggleGeneralInfoPanel() {
+        this.isGeneralInfoPanelVisible = !this.isGeneralInfoPanelVisible;
         this.cdr.detectChanges();
     }
 
-    handleTagsPanelState(tagPanelState: boolean) {
-        this.tagsPanelState = tagPanelState;
+    toggleTagsPanel(tagPanelState: boolean) {
+        this.isTagPanelVisible = tagPanelState;
         this.cdr.detectChanges();
     }
 
-    handleCategoriesPanelState(categoriesPanelState: boolean) {
-        this.categoriesPanelState = categoriesPanelState;
+    handleCategoriesPanelState(isCategoriesPanelVisible: boolean) {
+        this.isCategoriesPanelVisible = isCategoriesPanelVisible;
         this.cdr.detectChanges();
     }
 
+    hasTags(): boolean {
+       return !(this.tags.length > 0) && !this.editableTags;
+    }
+
+    hasCategories(): boolean {
+        return !(this.categories.length > 0) && !this.editableCategories;
+    }
+
+    get canToggleEdit(): boolean {
```

Comment:
Please use hasToggleEdit() instead of canToggleEdit().
Please replace where you have used the 'can' in the method.